### DISCUSSION TITLE
Access a created store again

### DISCUSSION THREAD

**Discussion by CodingRuo** [vuejs/pinia#2297] [Help and Questions]
Hy! :) 

I have a question, a definstore allows you to create a store. But how is that if the store was already created once, can I then access the store with the same function definstore when I pass the name, not or?

I wouldn't have found anything in the documentation. 

LG :)

**Comment by posva** [comment]
no, each store defined must have a unique name

**Reply by CodingRuo** [reply]
Hy! yes I know that :), but that was not my question. My question is, can the definestore function call a store that is already created programmatically?

for example i created a store with the name test is it possible then definestore(test, {} ) to call it again like this? Or is this not foreseen in pinia?

**Reply by posva** [reply]
no, that's not possible, you cannot "call" again a store

**Reply by CodingRuo** [reply]
No that's not quite true, I found the answer myself in the source code. You can see here that it is already possible, because all stores are stored in an array and if the id has already been instantiated, you can call it again. It must also be said that this is only possible when the store is created and active, if I switch from one page to another the store is of course gone. But thanks anyway! Ps : Here is the code snippet.
` if (!pinia._s.has(id)) {
      // creating the store registers it in pinia._s
      if (isSetupStore) {
        createSetupStore(id, setup, options, pinia)
      } else {
        createOptionsStore(id, options as any, pinia)
      }

      if (__DEV__) {
        // @ts-expect-error: not the right inferred type
        useStore._pinia = pinia
      }
    }

    const store: StoreGeneric = pinia._s.get(id)!
`

**Reply by posva** [reply]
not sure what you understand by "calling" a store but stores are created once and then reused. Also do not rely on internal properties like `_s` as they could potentially change 🙂

**Reply by CodingRuo** [reply]
Hi Eduardo,

We are working on a complex software application where we need to use the same logic over and over again in different stores. To facilitate this, I have developed a class that instantiates these stores for us. This allows us to work with pre-built functionality in the stores.

Since we instantiate many of these stores in the main layout and keep them active during runtime, the question arises how we can access a store that has already been instantiated. The solution to this is the defineStore() method. With this method, we can retrieve the existing store by specifying its ID and an empty object. However, this only works if the store already exists and has been instantiated.

As an alternative solution, we have also developed a static class in which all instantiated stores are stored. Using a get function, we can then retrieve the respective store as soon as it has been created.

Best regards,
René
